addons/purchase/views/purchase_views.xml

```diff
@@ -814,21 +823,91 @@
             </tree>
         </field>
     </record>
 
+    <record id="purchase_history_pivot_view" model="ir.ui.view">
+         <field name="name">purchase.history.pivot</field>
+         <field name="model">purchase.order.line</field>
+         <field name="arch" type="xml">
+             <pivot>
+                <field name="price_total" string="Total purchased amount" type="measure"/>
+                <field name="product_uom_qty" string="Total Qty purchased" type="measure"/>
+             </pivot>
+         </field>
+    </record>
+
+     <record id="purchase_history_graph_view" model="ir.ui.view">
+         <field name="name">purchase.history.graph</field>
+         <field name="model">purchase.order.line</field>
+         <field name="arch" type="xml">
+             <graph string="purchase Analysis" type="line" sample="1">
+                <field name="date_order" interval="month"/>
+                <field name="product_uom_qty" type="measure"/>
+             </graph>
+         </field>
+    </record>
+
+    <record id="purchase_history_kanban_view" model="ir.ui.view">
+        <field name="name">purchase.history.kanban</field>
+        <field name="model">purchase.order.line</field>
+        <field name="arch" type="xml">
+            <kanban class="o_kanban_mobile" action="action_open_order" type="object" sample="1" quick_create="false">
+                <field name="name"/>
+                <field name="order_id"/>
+                <field name="partner_id"/>
+                <field name="date_approve"/>
+                <templates>
+                    <t t-name="kanban-box">
+                        <div t-attf-class="oe_kanban_card oe_kanban_global_click">
+                            <div class="o_kanban_record_top mb16">
+                                <div class="o_kanban_record_headings mt4">
+                                    <strong class="o_kanban_record_title">
+                                        <span t-out="record.partner_id.value"/>
+                                    </strong>
+                                </div>
+                                <strong>
+                                    <field name="order_id"/>
+                                </strong>
+                            </div>
+                            <div class="o_kanban_record_bottom">
+                                <div class="oe_kanban_bottom_left text-muted">
+                                    <field name="date_approve"/>
+                                </div>
+                                <div class="oe_kanban_bottom_right">
+                                    <field name="price_unit"/>
+                                </div>
+                            </div>
+                        </div>
+                    </t>
+                </templates>
+            </kanban>
+        </field>
+    </record>
+
     <record id="action_purchase_history" model="ir.actions.act_window">
```

Comment:
Your change will also impact this button action 
![image](https://github.com/odoo/odoo/assets/77771734/a5d3860c-89dd-449a-a819-bb271fa9238f)

Did you check with PO if they are okay with this (including purchase PO)

Reply:
Yes, I have checked with po they agree with the changes.